Contest_API.md

```diff
@@ -1240,6 +1266,77 @@ Returned data:
 {"id":"nicky","username":"Nicky"}
 ```
 
+### Access
+
+Information on which endpoints and properties are visible to the current client, and what [capabilities](#capabilities)
+this client has access to or can perform.
+
+The following endpoint is associated with access:
+
+| Endpoint                | Mime-type        | Required? | Description
+| :---------------------- | :--------------- | :-------- | :----------
+| `/contests/<id>/access` | application/json | yes       | JSON object representing the current client's access with properties as defined in the table below.
+
+Properties of access objects:
+
+| Name         | Type                      | Required? | Nullable? | Description
+| :----------- | :------------------------ | :-------- | :-------- | :----------
+| capabilities | array of string           | no        | yes       | An array of [capabilities](#capabilities) that the current client has.
+| endpoints    | array of endpoint objects | yes       | no        | An array of endpoint objects that are visible to the current client, as described below.
+
+Properties of endpoint objects:
+
+| Name         | Type            | Required? | Nullable? | Description
+| :----------- | :-------------- | :-------- | :-------- | :----------
+| type         | string          | yes       | no        | The type of the endpoint, e.g. "problems". See table in [Notification format](#notification-format) for the list of types.
+| properties   | array of string | yes       | no        | An array of supported properties that the current client has visibility to.
+
+This endpoint provides information about what is accessible to a specific
+client in a live contest, and hence will not exist in a contest archive.
+
+This information is provided so that clients know what endpoints are available,
+what notifications may happen, and what capabilities they have, regardless
+of whether objects currently exist or the capability is currently active.
+For instance, a client logged in with a team account would see the problems type and
+team_submit capability before a contest starts, even through they cannot
+see any problems nor submit yet.
+Clients are not expected to call this endpoint more than once
+since the response would not normally change during a contest.
```

Comment:
Maybe better to change "would" to "should" or even "must"?

Reply:
Definitely not "must".

Reply:
Agreed. If you somehow make a mistake with setting up capabilities for a user you should be able to fix that mistake without being illegal.

Reply:
Ditto, not 'must'. If you felt strongly I'd be ok with 'should', but otherwise I think this is ok.

Reply:
Ok, I agree that "must" is too strong, but I think that changing it to "should" would be better, so that clients also have a soft guarantee that they don't need to query this endpoint more than once.

Reply:
Ok, changed.